symbulate/random_variables.py

```diff
@@ -5,30 +5,55 @@
 from .utils import is_scalar, is_vector, get_dimension
 
 class RV:
+    """Defines a random variable.
+    
+    A random variable is a function defined on a probability space.
+    Simulating a random variable is a two-step process: first,
+    a draw is taken from the underlying probability space; then,
+    the function is applied to that draw.
+
+    Args:
+      probSpace (ProbabilitySpace): the underlying probability space
+        of the random variable.
+      fun (function, optional): a function that maps draws from the 
+        probability space to numbers. (By default, the function is the 
+        identity function. For named distributions, a draw from the
+        underlying probability space is the value of the random
+        variable itself, which is why the identity function is the 
+        most frequently used.)
+
+    Attributes:
+      probSpace (ProbabilitySpace): the underlying probability space
+        of the random variable.
+      fun (function): a function that maps draws from the probability
+        space to numbers.
+
+    Examples:
+      P = Normal(0, 1)
```

Comment:
I would start with a simple example without the identity
P = BoxModel([0, 1], size = 5) # a single draw is a sequence of 0s and 1s, e.g., 0, 0, 1, 0, 1
X = RV(P, sum) # counts the number of 1s in the draw from P, e.g. 2

X = RV(Normal(0, 1))